lemur/plugins/lemur_gcp/tests/test_plugin.py

```diff
@@ -0,0 +1,89 @@
+EwJVUzETMBEGA1UECAwKQ2FsaWZvcm5pYTESMBAGA1UEBwwJTG9zIEdhdG9zMFkw
+EwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEVHkf6rYbpV1M7bPMFSbNxC6iHWm0HdvL
+baHIjh6FD9O4asxa5TOs8Z8Lbg3hTUgFamznF34J3oYfEjgTxEO40aMkMCIwIAYD
+VR0RAQH/BBYwFIISYmxhaEBsb2NhbGhvc3QuY29tMAoGCCqGSM49BAMCA0kAMEYC
+IQDLz0FkXEkKyGXfkO0XQ6HwF0Tw+QirLNQDgrErZWmzbQIhAOiNDLODpdPzf+Aj
+fQ6tr8edUIDueTN/LEqoDMlUX9up
+-----END CERTIFICATE-----
+"""
+private_key = """
+-----BEGIN EC PRIVATE KEY-----
+MHcCAQEEIH/TlH1MLEwUgPpZqd1EP3+9q792r7GsmecRQe6CknV4oAoGCCqGSM49
+AwEHoUQDQgAEVHkf6rYbpV1M7bPMFSbNxC6iHWm0HdvLbaHIjh6FD9O4asxa5TOs
+8Z8Lbg3hTUgFamznF34J3oYfEjgTxEO40Q==
+-----END EC PRIVATE KEY-----
+"""
+cert_chain = """
+-----BEGIN CERTIFICATE-----
+MIIB7TCCAZSgAwIBAgIQBm3vFdgxR8e2GOGwpR+XTDAKBggqhkjOPQQDAjBiMRIw
+EAYDVQQDDAlsb2NhbGhvc3QxFjAUBgNVBAoMDUV4YW1wbGUsIEluYy4xCzAJBgNV
+BAYTAlVTMRMwEQYDVQQIDApDYWxpZm9ybmlhMRIwEAYDVQQHDAlMb3MgR2F0b3Mw
+HhcNMjIwODI2MTkyMjMxWhcNNDIwODI2MTkyMjMxWjBiMRIwEAYDVQQDDAlsb2Nh
+bGhvc3QxFjAUBgNVBAoMDUV4YW1wbGUsIEluYy4xCzAJBgNVBAYTAlVTMRMwEQYD
+VQQIDApDYWxpZm9ybmlhMRIwEAYDVQQHDAlMb3MgR2F0b3MwWTATBgcqhkjOPQIB
+BggqhkjOPQMBBwNCAAQhNX4vrw7MYlenuUfEU5TYvYgjHGeJfULwJeYomzMloKWQ
+Msb0aRUWuEJ9STvqDSbHffK/Rm5BXAr328mzpIwRoywwKjAPBgNVHRMBAf8EBTAD
+AQH/MBcGA1UdEQEB/wQNMAuCCWxvY2FsaG9zdDAKBggqhkjOPQQDAgNHADBEAiB7
+dmVGV4armOiIvo+cyuAN8PLr4mq4ByiVFWl9WQavpAIgRA0leVMbErRrz78EEZZR
+aNVFrNhMcvbKB0eqb5VHL90=
+-----END CERTIFICATE-----
+"""
+
+SUCCESS_INSERT_RESPONSE = {
+    'kind': 'compute#operation',
+    'id': '4927389014336055823',
+    'name': 'operation-1661211870499-5e6dd077012f4-9b6e5e0d-ddfc98d2',
+    'operationType': 'insert',
+    'targetLink': 'https://www.googleapis.com/compute/v1/projects/testubg-sandbox/global/sslCertificates/test-cert-1234',
+    'targetId': '8919843282434501135',
+    'status': 'RUNNING',
+    'user': 'lemur-test@test.iam.gserviceaccount.com',
+    'progress': 0,
+    'insertTime': '2022-08-22T16:44:32.218-07:00',
+    'startTime': '2022-08-22T16:44:32.231-07:00',
+}
+
+options = [{
+    'name': 'accountName',
+    'type': 'str',
+    'required': True,
+    'value': 'lemur-test'
+}]
+
+
+@mock.patch("lemur.plugins.lemur_gcp.plugin.GCPDestinationPlugin.get_option", return_value="test-account")
+@mock.patch("lemur.plugins.lemur_gcp.plugin.GCPDestinationPlugin._get_gcp_credentials", return_value=token)
+@mock.patch("lemur.plugins.lemur_gcp.plugin.GCPDestinationPlugin._insert_gcp_certificate", return_value=SUCCESS_INSERT_RESPONSE)
+def test_upload(mock_sslCertificates, mock_credentials, mock_gcp_acount_id):
+
+    assert GCPDestinationPlugin().upload(
+        name,
+        body,
+        private_key,
+        cert_chain,
+        options) == SUCCESS_INSERT_RESPONSE
+
+
+@mock.patch("lemur.plugins.lemur_gcp.plugin.GCPDestinationPlugin._get_gcp_credentials_from_vault", return_value="ya29.c.b0AXv0zTN36HtXN2cJolg9tAj0vGAOT29FF-WNxQzvPu")
+def test_get_gcp_credentials__from_vault(mock__get_gcp_credentials_from_vault):
```

Comment:
Two minor nits:
- This method has two consecutive underscores in it (?)
- Double and single quotes seem to be randomly used within this file for strings

Reply:
Its a naming convention I picked up from previous teams i've been on to give a descriptor to the test. 
It follows a convention of `test_{method name}__{descriptor for test}`. I use it in pytest where you often have many separate tests for the same method so you can quickly get a descriptor for the given test. 
eg:
```
test_get_gcp_credentials__from_vault()
test_get_gcp_credentials__with_service_account_creds()
test_get_gcp_credentials__exception_thrown_without_valid_creds()
```
This isn't something defined in the pep8 style guide and is just a habit i've continued with. I'm happy to switch it so that matches our coding style. 

To you second bullet point, you are right, I accidentally have some double underscores in string names, eg: `mock__get_gcp_credentials_from_vault`, this was not intentional.